components/Ripple/src/MDCRippleView.m

```diff
@@ -120,6 +120,13 @@ - (void)updateRippleStyle {
 
 - (void)cancelAllRipplesAnimated:(BOOL)animated completion:(MDCRippleCompletionBlock)completion {
   NSArray<CALayer *> *sublayers = [self.layer.sublayers copy];
+  // Call callbacks immediately if no sublayers existing
+  if (sublayers.count == 0) {
```

Comment:
did you find this optimization to be worth the additional code complexity? If it isn't animated it will go into the added completion block you provided and if there are no sublayers all the for iterations over the sublayers would just be skipped over

Reply:
No, agree! Let's remove it!